src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobMetadataStore.cs

```diff
@@ -122,13 +140,50 @@ await using (Stream stream = _recyclableMemoryStreamManager.GetStream(GetInstanc
         }, cancellationToken);
     }
 
+    // TODO: This should removed once we migrate everything and the global flag is turned on
     private BlockBlobClient GetInstanceBlockBlob(VersionedInstanceIdentifier versionedInstanceIdentifier)
     {
-        var blobName = $"{versionedInstanceIdentifier.StudyInstanceUid}/{versionedInstanceIdentifier.SeriesInstanceUid}/{versionedInstanceIdentifier.SopInstanceUid}_{versionedInstanceIdentifier.Version}_metadata.json";
+        string blobName;
+        if (_supportNewBlobFormatForNewService)
+        {
+            blobName = _nameWithPrefix.GetMetadataFileName(versionedInstanceIdentifier);
+        }
+        else
+        {
+            blobName = _nameWithUid.GetMetadataFileName(versionedInstanceIdentifier);
+        }
 
         return _container.GetBlockBlobClient(blobName);
     }
 
+    private BlockBlobClient[] GetInstanceBlockBlobs(VersionedInstanceIdentifier versionedInstanceIdentifier)
+    {
+        var clients = new List<BlockBlobClient>();
+
+        string blobName;
+
+        if (_supportNewBlobFormatForNewService)
```

Comment:
I suppose this is an optimization so that we don't have to dual write on new services?

Reply:
Yes